### PR TITLE
hide the button and menu items when not signed in - in the chat list header

### PR DESCRIPTION
**Pull request by seekdavidlee**
chatlist has the context of whether we are signed in so we are passing in the state from chatlist to chatlistheader

### CODE REVIEW

File: packages/mgt-chat/src/components/ChatListHeader/ChatListHeader.tsx [hunk 1]
```diff
@@ -47,20 +48,19 @@ export const ChatListHeader = (
...
     <div className={classes.headerContainer}>
       <div>
-        {buttonItems.map((buttonItem, index) => (
-          <Button key={index} className={classes.button} onClick={buttonItem.onClick}>
-            <div className={classes.buttonIcon}>
-              <Circle>{buttonItem.renderIcon()}</Circle>
-            </div>
-          </Button>
-        ))}
+        {props.isSignedIn === true &&
```
Comment (andrewDoing) [inline packages/mgt-chat/src/components/ChatListHeader/ChatListHeader.tsx:51]: `=== true` is redundant here and on line 63

File: packages/mgt-chat/src/components/ChatList/ChatList.tsx [hunk 2]
```diff
@@ -77,6 +78,7 @@ export const ChatList = ({
         const client = new StatefulGraphChatListClient(props.chatThreadsPerPage);
         setChatListClient(client);
         setChatListState(client.getState());
+        setIsSignedIn(true);
```
Comment (MattBillFred) [inline packages/mgt-chat/src/components/ChatList/ChatList.tsx:81]: Do we need to handle the situation where the provider's state changed from "SignedIn" to something else? For example if it switches to "SignedOut", we may want to call `setIsSignedIn(false)`.

File: packages/mgt-chat/src/components/ChatList/ChatList.tsx [hunk 3]
```diff
@@ -67,6 +67,7 @@ export const ChatList = ({
   const [chatListClient, setChatListClient] = useState<StatefulGraphChatListClient | undefined>();
   const [chatListState, setChatListState] = useState<GraphChatListClient | undefined>();
   const [menuItems, setMenuItems] = useState<ChatListMenuItem[]>(props.menuItems === undefined ? [] : props.menuItems);
+  const [isSignedIn, setIsSignedIn] = useState<boolean>(false);
```
Comment (plasne) [inline packages/mgt-chat/src/components/ChatList/ChatList.tsx:70]: Are we sure we don't have a re-render on sign-in anyway? I thought that changed the state of the chatListClient or chatListState anyway?

File: packages/mgt-chat/src/components/ChatList/ChatList.tsx (hunk 2 above)
Comment (seekdavidlee) [reply packages/mgt-chat/src/components/ChatList/ChatList.tsx:81]: Do we have a case for when the State is changed to SignedOut thou after a SignedIn? If a user explicitly signs out, I think it causes a redirect out to the Entra page.

File: packages/mgt-chat/src/components/ChatListHeader/ChatListHeader.tsx (hunk 1 above)
Comment (seekdavidlee) [reply packages/mgt-chat/src/components/ChatListHeader/ChatListHeader.tsx:51]: Without `=== true`, we are just verifying the property is set.

File: packages/mgt-chat/src/components/ChatList/ChatList.tsx (hunk 3 above)
Comment (seekdavidlee) [reply packages/mgt-chat/src/components/ChatList/ChatList.tsx:70]: Not sure where you are getting at, but here we are just setting the initial value, which is following the pattern for all other state entities, no?

File: packages/mgt-chat/src/components/ChatList/ChatList.tsx (hunk 2 above)
Comment (MattBillFred) [reply packages/mgt-chat/src/components/ChatList/ChatList.tsx:81]: I don't have a specific use case for this. I just wasn't sure if there would ever be a case where the provider's state would change and the component's state be out of sync. I can't really think of a way to test this, so I will close out this comment.

**Review by TechnicallyWilliams — APPROVED** [review]

**Comment by seekdavidlee** [comment]
I will create a new PR

**Pull request closed without merging**